core/src/main/scala/org/apache/spark/sql/eventhubs/EventHubsSourceProvider.scala

```diff
@@ -156,7 +156,8 @@ private[sql] object EventHubsSourceProvider extends Serializable {
         StructField("sequenceNumber", LongType),
         StructField("enqueuedTime", TimestampType),
         StructField("publisher", StringType),
-        StructField("partitionKey", StringType)
+        StructField("partitionKey", StringType),
+        StructField("properties", MapType(StringType, StringType), nullable = true)
```

Comment:
Is it possible to provide the below different types of Schemas
1) Schema without properties
2) schema with properties - which serializes property values as strings
3) schema with properties - which does it this way - json way
and then, be able to choose schema - based on the Configuration. Perhaps, next step in this direction is to keep the schema part extensible in the spark adapter.

My concern here is that:
- all users - irrespective of whether they have just a single type of property value - need to deal with deserializing json; it will be good - if we have a way to support simple types easily and then provide ways to achieve these advanced serialization scenarios.
- Serialization errors, if exists, will pop-up - for the users who doesn't really care about properties for their scenario.